test/JDBC/expected/smalldatetime_date_cmp-vu-verify.out

```diff
@@ -0,0 +1,3863 @@
+GO
+~~START~~
+text
+0
+~~END~~
+
+SELECT set_config('debug_parallel_query', '0', false);
+GO
+~~START~~
+text
+off
+~~END~~
+
+SELECT set_config('babelfishpg_tsql.explain_costs', 'off', false)
+GO
+~~START~~
+text
+off
+~~END~~
+
+
+SET BABELFISH_SHOWPLAN_ALL ON
+GO
+
+-- seq scan
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-13' AS DATE);
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-13' AS DATE)
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: (smalldatetime_col <> '2023-06-13'::date)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 197.069 ms
+~~END~~
+
+
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-15' AS DATE);
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-15' AS DATE)
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: (smalldatetime_col <> '2023-06-15'::date)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.186 ms
+~~END~~
+
+
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-17' AS DATE);
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col <> CAST('2023-06-17' AS DATE)
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: (smalldatetime_col <> '2023-06-17'::date)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.178 ms
+~~END~~
+
+
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-13' AS DATE) <> smalldatetime_col;
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-13' AS DATE) <> smalldatetime_col
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: ('2023-06-13'::date <> smalldatetime_col)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 13.503 ms
+~~END~~
+
+
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-15' AS DATE) <> smalldatetime_col;
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-15' AS DATE) <> smalldatetime_col
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: ('2023-06-15'::date <> smalldatetime_col)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.213 ms
+~~END~~
+
+
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-17' AS DATE) <> smalldatetime_col;
+GO
+~~START~~
+text
+Query Text: SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE CAST('2023-06-17' AS DATE) <> smalldatetime_col
+Aggregate
+  ->  Seq Scan on smalldate_date_cmp_t1
+        Filter: ('2023-06-17'::date <> smalldatetime_col)
+~~END~~
+
+~~START~~
+text
+Babelfish T-SQL Batch Parsing Time: 0.174 ms
+~~END~~
+
+
+-- index scan
+SELECT COUNT(*) FROM smalldate_date_cmp_t1 WHERE smalldatetime_col < CAST('2023-06-14' AS DATE);
```

Comment:
can you also have order by clause and confirm it still picks index scan?

Reply:
is it possible to add test which involes join of two tables where inner side relation picks up index scan

Reply:
added tests
- ORDER BY clause.
- joins of 2 tables choosing index scan.